src/components/widget/BarcodeScanner/BarcodeScannerWidget.js

```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react';
+import BarcodeScanner, { BarcodeScannerResult } from './BarcodeScanner';
+
+function addBarcodeScanner(WrappedComponent) {
+  return class BarcodeScannerWidget extends Component {
+    constructor(props) {
+      super(props);
+
+      this.state = {
+        scanning: false,
+        result: null,
+        barcodeSelected: null,
+      };
+    }
```

Comment:
You can remove the constructor here and declare it as an instance variable, since it doesn't depend on props

```js
state = {
  // ...
};
```

Reply:
True, it's not needed in the HOC anymore.

Reply:
Not sure I understand - why? The state won't be shared between instances. It will be created again for each new instance (as would happen if you create it in the constructor)

Reply:
You still need to call `setState` instead of mutating `this.state` directly, of course

Reply:
Changing instance variable doesn't trigger render, so the wrapped component won't be re-rendered with the new value in props. So in this case if I'd change the value of `scanning` (which controls if scanner widget is visible or not), widget would stay opened.

Reply:
See my comment above, you still need to call `setState`. The comment was only concerning the instantiation of `this.state`. (Doing `this.state = ...;` in the constructor and declaring an instance variable `state = ...;` are equivalent)

Reply:
No, they're not :) Changing instance variables does not trigger render, as they're utside of React flow. 

From:
https://www.reddit.com/r/reactjs/comments/54gl72/why_use_state_vs_class_properties/

```
Setting a field directly on the component instance, like this.someField = 42, will not cause the component to actually re-render.
```